src/view/view.rs

```diff
@@ -110,4 +116,40 @@ impl Cx {
     pub fn add_pending_async(&mut self, id: Id) {
         self.pending_async.insert(id);
     }
+
+    /// Adds tree structure changes of the current element (and drains the mutations vec),
+    /// these should be applied when creating or changing a `Pod` of this element by calling
+    /// `cx.apply_children_tree_structure_mutations(pod.id())`
+    /// One needs to be careful with this, when recursively traversing the tree to avoid overwriting other mutations
+    pub fn mark_children_tree_structure_mutations(&mut self, mutations: &mut Vec<SpliceMutation>) {
+        self.current_children_tree_structure_mutations.clear();
+        self.current_children_tree_structure_mutations
+            .append(mutations);
+    }
+
+    // TODO(#160): Currently if there are no children added/mutated the tree will not be mutated.
+    //             Should the parent be inserted if there aren't any children? Should this happen elsewhere?
```

Comment:
I'm not sure that's a problem?

If I understand the current design, the only widget whose parent info isn't kept up to date is the root widget, which should be fine.

Reply:
> I'm not sure that's a problem?

Yeah it's no problem, I just like to write out thoughts while coding (before I forget them). And clean them up later. A good time is probably now.

> the only widget whose parent info isn't kept up to date is the root widget, which should be fine.

It should not even be necessary to update the root widget, since the root `Pod` is only created once, and never changes its id.
The children of that root widget are updated though in the tree structure.